pandas/core/algorithms.py

```diff
@@ -1102,6 +1102,37 @@ def _get_score(at):
         return result
 
 
+def check_percentile(q: Union[float, Iterable[float]]) -> np.ndarray:
+    """
+    Validate percentiles (used by describe and quantile).
+
+    This function checks if the given float oriterable of floats is a valid percentile otherwise raises a ValueError.
+
+    Args
+    ----
+    q: float or iterable of floats
+        A single percentile or an iterable of percentiles.
+
+    Returns
+    -------
+    ndarray
+        An ndarray of the percentiles if valid.
+
+    Raises
+    ------
+    ValueError if percentiles are not in given interval([0, 1]).
+    """
+    msg = "percentiles should all be in the interval [0, 1]. " "Try {0} instead."
+    q_arr = np.asarray(q)
+    if q_arr.ndim == 0:
+        if not 0 <= q_arr <= 1:
+            raise ValueError(msg.format(q_arr / 100.0))
+    else:
+        if not all(0 <= qs <= 1 for qs in q_arr):
```

Comment:
for a large array this is going to be much less performant than `((0 <= q_arr) & (q_arr <= 1)).all()`